### PR TITLE
StaticFlags: Implement NO_MELEE_APPROACH

### PR DESCRIPTION
**Pull request by insunaa**
## 🍰 Pullrequest

This PR implements CreatureStaticFlags4::NO_MELEE_APPROACH

### CODE REVIEW

**Comment by insunaa** [comment]
This could also mean the exact opposite: Melee combat is disabled, but the Creature tries to run into melee range, similar to NO_MELEE_FLEE except it approaches instead of flees. I'm not sure

**Comment by Shauren** [comment]
The proper name of this flag is `No Melee (Approach)` - autoattacks are disabled but creatures chase as-if it wasnt

`No Melee (Flee)` - autoattacks are disabled and creature runs away (the critter behavior) there are no non-critter and non-pet creatures before cataclysm that use this flag (fleeing part of this flag is disabled/overriden if the creature is a pet (including pet of another creature, not just player) 

Cataclysm has one https://www.wowhead.com/npc=40168/corrupted-dryad

**Comment by insunaa** [comment]
Thanks a lot, I'll update the code!

File: src/game/AI/BaseAI/CreatureAI.cpp [hunk 1]
```diff
@@ -35,6 +35,7 @@ CreatureAI::CreatureAI(Creature* creature, uint32 combatActions) :
 {
     m_dismountOnAggro = !(m_creature->GetCreatureInfo()->CreatureTypeFlags & CREATURE_TYPEFLAGS_MOUNTED_COMBAT);
     SetMeleeEnabled(!m_creature->GetSettings().HasFlag(CreatureStaticFlags::NO_MELEE_FLEE));
+    SetMeleeEnabled(!m_creature->GetSettings().HasFlag(CreatureStaticFlags4::NO_MELEE_APPROACH));
```
Comment (killerwife) [inline src/game/AI/BaseAI/CreatureAI.cpp:38]: Set it just once lol

**Pull request merged**